pkg/controller/migmigration/validation.go

```diff
@@ -290,35 +291,41 @@ func ensureRegistryHealth(c k8sclient.Client, migration *migapi.MigMigration) (i
 			return nEnsured, message, nil
 		}
 
-		registryStatusUnhealthy, podObj := isRegistryPodUnHealthy(registryPods)
-
+		registryStatusUnhealthy, podObj, state := isRegistryPodUnHealthy(registryPods)
 		if !registryStatusUnhealthy {
 			nEnsured++
 		} else {
 			unHealthyPod = podObj
 			unHealthyClusterName = cluster.ObjectMeta.Name
+			reason = state
 		}
 	}
 
 	if nEnsured != 2 {
-		message := fmt.Sprintf("Migration Registry Pod %s/%s is in unhealthy state on cluster %s",
-			unHealthyPod.Namespace, unHealthyPod.Name, unHealthyClusterName)
+		message := fmt.Sprintf("Migration Registry Pod %s/%s is in unhealthy state on cluster %s, the Pod is in %s state",
+			unHealthyPod.Namespace, unHealthyPod.Name, unHealthyClusterName, reason)
+		if reason == "ImagePullBackOff" {
+			return nEnsured, message, errors.New(reason)
+		}
 		return nEnsured, message, nil
 	}
 
 	return nEnsured, "", nil
 }
 
-func isRegistryPodUnHealthy(registryPods corev1.PodList) (bool, corev1.Pod) {
+func isRegistryPodUnHealthy(registryPods corev1.PodList) (bool, corev1.Pod, string) {
```

Comment:
@JaydipGabani would you mind adding a comment above this func describing the expected function output now that we have this third string return value?

Reply:
@djwhatle Done.